gateway/analytics/segment.go

```diff
@@ -23,37 +22,28 @@ func New() *Segment {
 }
 
 func (s *Segment) Identify(ctx *types.APIContext) {
-	if s.Client == nil || ctx == nil || ctx.UserEmail == "" || ctx.OrgID == "" ||
+	if s.Client == nil || ctx == nil || ctx.UserID == "" || ctx.OrgID == "" ||
 		!appconfig.Get().AnalyticsTracking() {
 		return
 	}
 
 	_ = s.Client.Enqueue(analytics.Identify{
-		// Segment recommends using an unique id for user id
-		// However we use the e-mail to avoid having to associate the
-		// user id with the e-mail.
-		UserId:      ctx.UserEmail,
```

Comment:
The impact of removing this condition is difficult to fully assess. For context, this information is stored in the database across sessions and reviews, where the ID is validated against the user context during the review flow.

Given the scope, this change will require extensive testing to prevent breaking functionality.

I propose an alternative approach: Keep the current configuration intact and gradually refactor the codebase to eliminate UserId dependencies. This incremental strategy will be easier to test and validate compared to implementing everything in a single PR.